models/Asset/Service.php

```diff
@@ -166,18 +171,98 @@ public function copyContents($target, $source)
         return $target;
     }
 
+
     /**
-     * @param  Asset $asset
+     * @param $asset
+     * @param null $fields
+     * @param null $requestedLanguage
+     * @param array $params
      *
-     * @return $this
+     * @return array
      */
-    public static function gridAssetData($asset)
+    public static function gridAssetData($asset, $fields = null, $requestedLanguage = null, $params = [])
     {
         $data = Element\Service::gridElementData($asset);
 
+        if ($asset instanceof Asset && !empty($fields)) {
+
+            $data = [
+                'id' => $asset->getid(),
+                'type' => $asset->getType(),
+                'fullpath' => $asset->getRealFullPath(),
+                'filename' => $asset->getKey(),
+                'creationDate' => $asset->getCreationDate(),
+                'modificationDate' => $asset->getModificationDate(),
+                'idPath' => Element\Service::getIdPath($asset),
+            ];
+
+            $requestedLanguage = str_replace("default","", $requestedLanguage);
+
+            foreach ($fields as $field) {
+                if ($field == "preview") {
```

Comment:
Please note that a metadata key can be called "preview". Such a key cannot be added to the grid!

Reply:
same for all system columns